src/components/MoneyRequestAmountInput.tsx

```diff
@@ -216,10 +241,7 @@ function MoneyRequestAmountInput(
                 end: frontendAmount.length,
             });
         }
-
-        // we want to re-initialize the state only when the amount changes
-        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
-    }, [amount, shouldKeepUserInput]);
+    }, [amount, currency, formatAmountOnBlur, shouldKeepUserInput, onFormatAmount, shouldResetAmount, onResetAmount, currentAmount]);
```

Comment:
Hey 👋 It looks like this change introduced a bug (check the [revert PR description](https://github.com/Expensify/App/pull/45328#issue-2405131554))

cc @mountiny

Reply:
I tried to bring those lines back

```
        // we want to re-initialize the state only when the amount changes
        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
    }, [amount, shouldKeepUserInput]);
```

and the input is working again, so a fix should be pretty easy :)

Reply:
Heads up we ended up reverting this PR because of the above mentioned critical issue.

We opted for a revert to make sure there are no other unwanted regressions

@kaushiktd can you raise a new PR and make sure to test for the input working well